Shipr-Helm/templates/shipr-payment-deployment.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ Values.payment.name }}
+  labels:
+    app: {{ Values.payment.name }}
+spec:
+  replicas: {{ Values.payment.replicas }}
+  selector:
+    matchLabels:
+      app: {{ Values.payment.name }}
+  template:
+    metadata:
+      labels:
+        {{- with .Values.payment.labels }}
+        {{- toYaml . | nindent 6 }}
+        {{- end }}
+    spec:
+      containers:
+      - name: {{ Values.payment.name }}
+        image: {{ Values.payment.repository }}:{{ Values.payment.tag }}
+        ports:
+        - containerPort: {{ Values.payment.containerPort }}
+        env:
+          - name: {{ Values.payment.env.host-name }}
+            value: {{ Values.payment.env.host-value }}
+          - name: {{ Values.payment.env.port-name }}
+            value: {{ Values.payment.env.port-value }}
+          - name: {{ Values.payment.env.db-name }}
+            value: {{ Values.payment.env.db-value }}
+          - name: {{ Values.payment.env.url-name }}
+            value: {{ Values.payment.env.url-value }}
+          - name: {{ Values.payment.env.fee-name }}
+            value: {{ Values.payment.env.fee-value }}
+        livenessProbe:
+          httpGet:
+            path: /
+            port: {{ Values.payment.containerPort }}
+          initialDelaySeconds: 10
+          periodSeconds: 10
+        readinessProbe:
+          httpGet:
+            path: /
+            port: {{ Values.payment.containerPort }}
+          initialDelaySeconds: 5
+          periodSeconds: 5
+        startupProbe:
+          httpGet:
+            path: /
+            port: {{ Values.payment.containerPort }}
+          initialDelaySeconds: 15
+          periodSeconds: 5
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ Values.payment.name }}
+spec:
+  selector:
+    app: {{ Values.payment.name }}
+  ports:
+    - protocol: TCP
+      port: {{ Values.payment.service.port }}
+      targetPort: {{ Values.payment.service.targetport }}
+      nodePort: {{ Values.payment.service.nodePort }}
+  type: {{ Values.payment.type }}
```

Comment:
If user chooses ClusterIP as the type of service, what value will the `nodePort` be?

Reply:
Applied the check here